scripts/kernel_bump.sh

```diff
@@ -74,67 +79,89 @@ cleanup()
 
 init()
 {
+	src_file="$(readlink -f "${0}")"
+	src_dir="${src_file%%"${src_file##*'/'}"}"
 	initial_branch="$(git rev-parse --abbrev-ref HEAD)"
 	initial_commitish="$(git rev-parse HEAD)"
 
-	trap cleanup EXIT HUP INT QUIT ABRT ALRM TERM
-}
+	if [ -n "$(git status --porcelain | grep -v '^?? .*')" ]; then
+		echo 'Git respository not in a clean state, will not continue.'
+		exit 1
+	fi
 
-do_source_target()
-{
-	_target_dir="${1:?Missing argument to function}"
-	_op="${2:?Missing argument to function}"
+	if [ -n "${src_dir##*'/scripts/'}" ]; then
+		echo "This script '${src_file}' is not in the scripts subdirectory, this is unexpected, cannot continue."
+		exit 1
+	fi
+
+	source_version="${source_version#v}"
+	target_version="${target_version#v}"
 
+	trap cleanup EXIT HUP INT QUIT ABRT ALRM TERM
 }
 
 bump_kernel()
 {
-	platform_name="${platform_name##*'/'}"
-
-	if [ -z "${platform_name:-}" ]; then
-		platform_name="${PWD##*'/'}"
+	if [ -z "${platform_name}" ] || \
+	   [ -s "${PWD}/image" ]; then
+		platform_name="${PWD}"
 	fi
+	platform_name="${platform_name##*'/'}"
 
-	if [ "${PWD##*'/'}" = "${platform_name}" ]; then
-		_target_dir='./'
-	else
-		_target_dir="target/linux/${platform_name}"
-	fi
+	_target_dir="${src_dir}/../target/linux/${platform_name}"
 
 	if [ ! -d "${_target_dir}/image" ]; then
-		e_err 'Cannot find target linux directory.'
+		e_err "Cannot find target linux directory '${_target_dir:-not defined}'. Not in a platform directory, or -p not set."
 		exit 1
 	fi
 
 	git switch --force-create '__openwrt_kernel_files_mover'
 
-	for _path in "${_target_dir}/"*; do
-		if [ ! -s "${_path}" ] || \
-		   [ "${_path}" = "${_path%%"-${source_version}"}" ]; then
+	if [ "${config_only:-false}" != 'true' ]; then
+		for _path in $(git ls-tree -d -r --name-only '__openwrt_kernel_files_mover' "${_target_dir}" |
+			       sed -n "s|^\(.*-${source_version}\).*|\1|p" |
+			       sort -u); do
+			if [ ! -s "${_path}" ] || \
+			   [ "${_path}" = "${_path%%"-${source_version}"}" ]; then
+				continue
```

Comment:
This testing doesn't work for Windows systems (TartoiseGIT - GIT Bash here). For Windows, folders always have size 0.